src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -677,9 +703,14 @@ int D4NFilterObject::D4NFilterReadOp::prepare(optional_yield y, const DoutPrefix
   //set a flag to show that incoming instance has no version specified
   bool is_latest_version = true;
   if (source->have_instance()) {
-    is_latest_version = false;
+    is_latest_version = false; // what about the case where the version supplied is the latest version? -Sam
```

Comment:
Yes, thats ok, in that case, the block containing the head version will not be set to the latest version. Otherwise the instance block for that version will be present in the cache, or will be cached. In case of the write-back cache, every new object written will be the latest version. This situation will only arise in case of read only workflows, and I think the user will read the latest version without supplying the version id. In any case, I plan to fix this corner case. please don't make any changes to the readop::prepare() for this.

Reply:
Ok, I will remove my comment since you are already looking into this. I haven't made any changes to `prepare` in regards to this corner case and I've updated the `delete_obj` logic to store a block instance to represent delete markers so the workflows should work as intended.

Reply:
Yes, I've confirmed these workflows complete successfully with the correct values in the directory.